sillsdev/languageforge-lexbox
# Login with Google button

Fix #522.

![image](https://github.com/sillsdev/languageforge-lexbox/assets/90762/8264acee-dd4d-4fc7-8651-6ed0b174900d)

To test:

Edit `deployment/local-dev/local.env` and add:

```plaintext
GOOGLE_OAUTH_CLIENT_ID=something.apps.googleusercontent.com
GOOGLE_OAUTH_CLIENT_SECRET=do-not-check-this-in
```

Make sure that the Google OAuth client ID and secret that you use have http://localhost/api/login/signin-google as an authorized redirect URL. (Go to https://console.cloud.google.com/apis/ and edit the project credentials to add additional redirect URLs).

Go to the login page, and check that the "Login with Google" button appears. Click on it, and you should be taken to the new-user registration page, with your name and email address already filled in. Choose a password and register, and optionally verify your email address by going to `localhost:1080` to click the verification link. Then log out and click the "Login with Google" button again. This time, you should be taken straight to your home page, having logged in successfully with the newly-created user account.
